snapshots/overlay/overlay_test.go

```diff
@@ -173,21 +175,14 @@ func testOverlayOverlayMount(t *testing.T, newSnapshotter testsuite.SnapshotterF
 		upper = "upperdir=" + filepath.Join(bp, "fs")
 		lower = "lowerdir=" + getParents(ctx, o, root, "/tmp/layer2")[0]
 	)
-
-	expected := []string{
-		"index=off",
-	}
-	if userxattr, err := NeedsUserXAttr(root); err != nil {
-		t.Fatal(err)
-	} else if userxattr {
-		expected = append(expected, "userxattr")
-	}
-	expected = append(expected, []string{
-		work,
-		upper,
-		lower,
-	}...)
-	for i, v := range expected {
+	var slice []string
+	indexOffOnce.Do(getIndexOff)
+	if indexOff {
+		slice = []string{"index=off", work, upper, lower}
+	} else {
+		slice = []string{work, upper, lower}
+	}
```

Comment:
I'd make use of the slice, and append what's needed;
(assuming the check is implemented in `overlayutils`), something like

```suggestion
	if overlayutils.SupportsIndex() {
		expected = append(expected, "index=off")
	}
	expected = append(expected, work, upper, lower)
```

Or the reverse (remove if unsupported); (depends a bit on other suggestions, and what's cleaner/easier);

```suggestion
	expected := []string{"index=off", work, upper, lower}
	if !overlayutils.SupportsIndex() {
		expected = expected[1:]
	}
```